src/ShopifyApp/Models/Charge.php

```diff
@@ -169,30 +171,65 @@ public function remainingTrialDays()
      */
     public function remainingTrialDaysFromCancel()
     {
-        if (!$this->isTrial()) {
+        if ( ! $this->isTrial())
+        {
             return;
         }
 
         $cancelledDate = Carbon::parse($this->cancelled_on);
         $trialEndsDate = Carbon::parse($this->trial_ends_on);
 
         // Ensure cancelled date happened before the trial was supposed to end
-        if ($this->isCancelled() && $cancelledDate->lte($trialEndsDate)) {
+        if ($this->isCancelled() && $cancelledDate->lte($trialEndsDate))
+        {
             // Diffeence the two dates and subtract from the total trial days to get whats remaining
             return $this->trial_days - ($this->trial_days - $cancelledDate->diffInDays($trialEndsDate));
         }
 
         return 0;
     }
 
+    /**
+     * Returns the remaining days for recurring charge from cancellation date.
+     *
+     * @return int
+     */
+    public function remainingDaysAfterCancel()
+    {
+        if ( ! $this->isCancelled())
+        {
+            return false;
+        }
+
+        if ( ! $this->plan->isType(Plan::PLAN_RECURRING))
+        {
+            return false;
+        }
+
+
+        $cancelledDate = Carbon::parse($this->cancelled_on);
+
+        $periods = (int)(Carbon::parse($this->activated_on)->diffInDays(Carbon::today()) / 30);
+        $pastDaysInPeriod = Carbon::parse($this->activated_on)->addDays(30 * $periods)->diffInDays(Carbon::today());
+
+        if ($pastDaysInPeriod == 0 && $cancelledDate->lt(Carbon::today()))
```

Comment:
@ohmybrew i dont know if this special-case makes sense right now. Maybe somebody can validate this. The idea is when the user cancel on the date when a new 30 days period begins. than a new period of 30 days should be active before cancel comes into play